core/src/main/scala/org/apache/spark/util/ArrayReflect.java

```diff
@@ -0,0 +1,357 @@
+/*
+* Licensed to the Apache Software Foundation (ASF) under one or more
+* contributor license agreements.  See the NOTICE file distributed with
+* this work for additional information regarding copyright ownership.
+* The ASF licenses this file to You under the Apache License, Version 2.0
+* (the "License"); you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+package org.apache.spark.util;
+
+/**
+ * Alternative implementation of {@link java.lang.reflect.Array} that is more performant
+ * over calling to native C in many cases. Based on code from
+ * <a href=https://bugs.openjdk.java.net/browse/JDK-8051447>an open JDK ticket</a>.
+ */
+public class ArrayReflect {
```

Comment:
We only use `getLength` and `get`. In the name of minimizing the chunk of code that needs to be added, I suppose we can just implement those two methods?